optimum/intel/openvino/modeling_base.py

```diff
@@ -229,6 +236,20 @@ def _create_infer_request(self, model):
         compiled_model = core.compile_model(model, self._device, self.ov_config)
         return compiled_model.create_infer_request()
 
+    def _reshape(self, model: openvino.runtime.Model, batch_size: int, sequence_length: int):
+        shapes = {}
+        for inputs in model.inputs:
+            shapes[inputs] = inputs.get_partial_shape()
+            shapes[inputs][0] = batch_size
+            shapes[inputs][1] = sequence_length
+        model.reshape(shapes)
+        return model
+
+    def reshape(self, batch_size: int, sequence_length: int):
+        self.is_dynamic = True if batch_size == -1 and sequence_length == -1 else False
+        self.model = self._reshape(self.model, batch_size, sequence_length)
+        self.request = self._create_infer_request(self.model)
+
     def _ensure_supported_device(self, device: str = None):
```

Comment:
Now that both CPU and GPU are supported, we should also support AUTO, MULTI and HETERO. See https://docs.openvino.ai/latest/openvino_docs_OV_UG_Working_with_devices.html 
Device names can then be for example MULTI:CPU,GPU, or just AUTO (which will automatically choose the best device), or AUTO:CPU,GPU (for example to specify that AUTO is allowed to choose from CPU and GPU, but not VPU even if it is available on the system) or AUTO:GPU,CPU, etc.

Reply:
Good point, I will add these to the list of the supported devices